solver/jobs.go

```diff
@@ -511,7 +519,8 @@ func (jl *Solver) deleteIfUnreferenced(k digest.Digest, st *state) {
 
 func (j *Job) Build(ctx context.Context, e Edge) (CachedResultWithProvenance, error) {
 	if span := trace.SpanFromContext(ctx); span.SpanContext().IsValid() {
-		j.span = span
+		// CompareAndSwap ensures we store the first span and ignore updates.
+		j.span.CompareAndSwap(nil, &span)
```

Comment:
How can this ever be nil if it is always initialized in `NewJob()`? I think the logic that `NewJob()` sets a no-op tracer, first `Build` may set actual tracer from context, and now if there are any new builds then these are ignored.